src/test/vscodeTest/suite/index.ts

```diff
@@ -11,27 +11,14 @@ export function run(): Promise<void> {
 
   const testsRoot = path.resolve(__dirname, '..');
 
-  return new Promise((c, e) => {
-    glob('**/**.test.js', { cwd: testsRoot }, (err, files) => {
-      if (err) {
-        return e(err);
-      }
+  const files = await glob('**/**.test.js', { cwd: testsRoot });
+  // Add files to the test suite
+  files.forEach(f => mocha.addFile(path.resolve(testsRoot, f)));
 
-      // Add files to the test suite
-      files.forEach(f => mocha.addFile(path.resolve(testsRoot, f)));
-
-      try {
-        // Run the mocha test
-        mocha.run(failures => {
-          if (failures > 0) {
-            e(new Error(`${failures} tests failed.`));
-          } else {
-            c();
-          }
-        });
-      } catch (err) {
-        e(err);
-      }
-    });
+  // Run the mocha test
+  mocha.run(failures => {
```

Comment:
`mocha.run` will return before the callback is run, which means the program will exit without reporting any passes or failures. To get around this, you need to wrap it in a Promise and await that promise.

See https://github.com/redhat-developer/vscode-quarkus/pull/648/files#diff-5116d4bcabdc6045aa26cc1afdd56ab3155276c7d69ff9b3be07b9fb157da247R19

Reply:
right. I still suck at Promises ;-)